### PR TITLE
#213 Add new fields for event DB tables

### LINKED ISSUE #213: Add new fields for event DB tables
Tables `animal_event_general`, `animal_event_medical_record`, `animal_event_found` and `animal_event_given_away` should include fields `author` (VARCHAR(255)) and `mod_time` (TIMESTAMP).
Update tables accordingly.

### PR DESCRIPTION
**Pull request by LukasKri**
Hey @ignasposka, I've added new fields (`author` and `mod_time`) to tables `animal_event_general`, `animal_event_medical_record`, `animal_event_found` and `animal_event_given_away`. Was I supposed to create `mod_time` triggers and add fake data as well? If no, just let me know and I will delete them.

Resolves #213.

### CODE REVIEW

**Comment by ignasposka** [comment]
> Hey @ignasposka, I've added new fields (`author` and `mod_time`) to tables `animal_event_general`, `animal_event_medical_record`, `animal_event_found` and `animal_event_given_away`. Was I supposed to create `mod_time` triggers and add fake data as well? If no, just let me know and I will delete them.
> 
> Resolves #213.

Yes, please add triggers and add some fake data.

**Comment by LukasKri** [comment]
I’ve already added them in advance, let me know if that satisfies you.

File: database/1-schema.sql [hunk 1]
```diff
@@ -284,7 +284,9 @@ CREATE TABLE animal_event_general (
     type event,
     expenses NUMERIC,
     date_time TIMESTAMP,
-    comments TEXT
+    comments TEXT,
+    author VARCHAR(255),
```
Comment (JustasBelevicius) [inline database/1-schema.sql:288]: This should be a foreign key to the app_user table, same in other places. Probably wasn't mentioned in the spec, sorry for that!

File: database/3-fake-data.sql [hunk 2]
```diff
@@ -61,51 +61,51 @@ INSERT INTO animal_microchip
...
-(5, '3', 1.00, '2018-07-28', 'Pasikeitė laikymo vietos adresas.'),
-(5, '3', 1.00, '2018-07-29', 'Pasikeitė laikymo vietos adresas.'),
-(5, '3', 1.00, '2018-07-30', 'Pasikeitė laikymo vietos adresas.'),
-(5, '3', 1.00, '2018-07-31', 'Pasikeitė laikymo vietos adresas.'),
-(5, '3', 1.00, '2018-08-01', 'Pasikeitė laikymo vietos adresas.'),
-(5, '3', 1.00, '2018-08-02', 'Pasikeitė laikymo vietos adresas.'),
-(5, '3', 1.00, '2018-08-03', 'Pasikeitė laikymo vietos adresas.'),
-(5, '3', 1.00, '2018-08-04', 'Pasikeitė laikymo vietos adresas.');
+(animal, type, expenses, date_time, comments, author) VALUES
+(2, '1', 109.03, '2020-07-01', 'Registration!', 'Benas'),
```
Comment (JustasBelevicius) [inline database/3-fake-data.sql:65]: Once you change the field to be a foreign key, update these as well. thanks! Same for other tables

**Review by JustasBelevicius — CHANGES_REQUESTED** [review]

File: database/1-schema.sql (hunk 1 above)
Comment (LukasKri) [reply database/1-schema.sql]: Hi @JustasBelevicius, my apologies, I should have known that myself. Have a look when you have a moment and please let me know what you think.

**Review by JustasBelevicius — APPROVED** [review]
Looks good to me! Only the fix needed, and then you can merge. Well done!

**Review by AzisK — APPROVED** [review]
GJ

**Comment by AzisK** [comment]
For some reason tests are failing, please fix it

**Comment by LukasKri** [comment]
Thanks guys for your feedback!

I had to update a lot of files, because new mutations was added to the tables while I was updating them and therefore new tests of those mutations were failing. I believe everything is up to date right now, could you give a final check, please?

P. S. I tried to squash my commits before pushing and do not know if I didn't mess them up.

File: src/sql-queries/animalEventFound.ts [hunk 3]
```diff
@@ -13,30 +14,33 @@ export const getAnimalFoundEventsQuery = (): QueryConfig => {
...
+    const { date, ...otherFields } = input;
     return { dateTime: date, ...otherFields };
 }
 
-export const createAnimalFoundEventQuery =
-    (input: CreateAnimalFoundEventInput): QueryConfig =>
-        insert(table, snakeCaseKeys(dateToDateTime(input)))
-            .returning(returningFields).toParams();
-
+export const createAnimalFoundEventQuery = (
```
Comment (ignasposka) [inline src/sql-queries/animalEventFound.ts:41]: Formatting of this function looked better previously..

File: src/sql-queries/animalEventGivenAway.ts [hunk 4]
```diff
@@ -3,37 +3,43 @@ import { insert, update } from 'sql-bricks-postgres';
...
-const returnFields = 'id, former_owner_id, date_time as date, animal_id, reason';
-const dateToDateTime =
-    (input: CreateGivenAwayEventInput | UpdateGivenAwayEventInput) => {
-        const { date, ...inputWithoutDate } = input;
-        return { ...inputWithoutDate, dateTime: date };
-    };
+const returnFields =
+    'id, former_owner_id, date_time as date, animal_id, reason, author';
+const dateToDateTime = (
+    input: CreateGivenAwayEventInput | UpdateGivenAwayEventInput
```
Comment (ignasposka) [inline src/sql-queries/animalEventGivenAway.ts:9]: Same with formatting.

File: test/givenAwayEvent.graphql.test.ts [hunk 5]
```diff
@@ -25,14 +26,19 @@ describe('GraphQL animal given away event integration tests', () => {
...
                         }) ${givenAwayEventFields}
-                    }`
+                    }`,
             })
             .expect(200)
             .set('Authorization', `Bearer ${process.env.BEARER_TOKEN}`)
             .end((err, res) => {
                 if (err) return done(err);
-                const { body: { data: { createGivenAwayEvent } } } = res;
+                const {
```
Comment (ignasposka) [inline test/givenAwayEvent.graphql.test.ts:37]: Let's revert this formatting change.

File: test/givenAwayEvent.graphql.test.ts [hunk 6]
```diff
@@ -49,15 +55,20 @@ describe('GraphQL animal given away event integration tests', () => {
...
+                            author: "dhjbwau74a6"
                         }) ${givenAwayEventFields}
                     }`,
             })
             .expect(200)
             .set('Authorization', `Bearer ${process.env.BEARER_TOKEN}`)
             .end((err, res) => {
                 if (err) return done(err);
-                const { body: { data: { updateGivenAwayEvent } } } = res;
+                const {
```
Comment (ignasposka) [inline test/givenAwayEvent.graphql.test.ts:67]: Let's revert this formatting change.

**Review by ignasposka — CHANGES_REQUESTED** [review]
Just some formatting issues.

**Comment by LukasKri** [comment]
Hey @ignasposka, changes that were made by Prettier in `animalEventFound.ts`, `animalEventGivenAway.ts` and `givenAwayEvent.graphql.test.ts` files were reverted and because of previous merge some tests were failing again, so I had to make some changes in order to make them pass again.

**Review by ignasposka — APPROVED** [review]

**Pull request merged**